### PR TITLE
#441 - Demonstrating how to do direct inserts with the JdbcAggregateTemplate.

### PR DESCRIPTION
**Pull request by schauder**
See also: https://jira.spring.io/browse/DATAJDBC-282

### CODE REVIEW

File: jdbc/basics/src/test/java/example/springdata/jdbc/basics/simpleentity/SimpleEntityTests.java [hunk 1]
```diff
@@ -61,4 +61,13 @@ public void exerciseRepositoryForSimpleEntity() {
 		repository.delete(cars);
 		Output.list(repository.findAll(), "`Cars` is gone.");
 	}
+
+	@Test
+	public void directInsert(){
```
Comment (mp911de) [inline jdbc/basics/src/test/java/example/springdata/jdbc/basics/simpleentity/SimpleEntityTests.java]: Shouldn't we rather point out the simple-most approach which would be direct `JdbcAggregateOperations` usage?

File: jdbc/basics/src/main/java/example/springdata/jdbc/basics/simpleentity/CategoryConfiguration.java [hunk 2]
```diff
@@ -61,4 +72,17 @@
...
 	}
+
+	// the following bean definitions are only necessary for providing JdbcAggregateOperations for injection into WithInsertImpl.
+	@Bean
+	public NamedParameterJdbcOperations namedParameterJdbcOperations(JdbcOperations operations) {
+		return new NamedParameterJdbcTemplate(operations);
+	}
+
+	@Bean
+	public JdbcAggregateOperations jdbcAggregateOperations(ApplicationEventPublisher publisher,
```
[the remark this answers is not in the document]
Comment (sdeleuze) [reply jdbc/basics/src/main/java/example/springdata/jdbc/basics/simpleentity/CategoryConfiguration.java]: Big :+1: for that, I would expect the same behavior than with MongoDB, especially if we don't get `insert()`/`update()` at repository level.

[the remark this answers is not in the document]
Comment (schauder) [reply jdbc/basics/src/main/java/example/springdata/jdbc/basics/simpleentity/CategoryConfiguration.java]: Done.

File: jdbc/basics/src/test/java/example/springdata/jdbc/basics/simpleentity/SimpleEntityTests.java (hunk 1 above)
Comment (schauder) [reply jdbc/basics/src/test/java/example/springdata/jdbc/basics/simpleentity/SimpleEntityTests.java]: I'm not sure.
As the intense discussion about DATAJDBC-282 shows many people want the insert method in their repositories. 
With the custom method, they get just that. 

Also, I guess most Spring Data JDBC users come from Spring Data JPA and don't even know about templates. 

I'd rather leave it as it is.

**Comment by mp911de** [comment]
That's polished and merged now.

**Pull request closed without merging**